src/network/networkd-dhcp6.c

```diff
@@ -1057,12 +1060,17 @@ static int dhcp6_assign_delegated_prefix(Link *link,
 
         r = address_new(&address);
         if (r < 0)
-                return log_link_error_errno(link, r, "Could not allocate address: %m");
+                return log_link_error_errno(link, r, "Failed to allocate address for DHCPv6 delegated prefix: %m");
 
         address->in_addr.in6 = *prefix;
-        r = generate_ipv6_eui_64_address(link, &address->in_addr.in6);
-        if (r < 0)
-                return log_link_warning_errno(link, r, "Failed to generate EUI64 address for DHCPv6 acquired delegated prefix: %m");
+
+        if (!in_addr_is_null(AF_INET6, &link->network->dhcp6_delegation_prefix_token))
+                memcpy(((uint8_t *) &address->in_addr.in6) + 8, ((uint8_t *) &link->network->dhcp6_delegation_prefix_token) + 8, 8);
```

Comment:
`(uint8_t *) &address->in_addr.in6 + 8` → `address->in_addr.in6.s6_addr + 8`

i.e. avoid casts, use the official members instead.

similar for the right hand side

Reply:
OK. Thanks.